Source/RapyutaSimulationPlugins/Private/Drives/RobotVehicleMovementComponent.cpp

```diff
@@ -25,14 +26,182 @@ void URobotVehicleMovementComponent::UpdateMovement(float InDeltaTime)
     DesiredRotation = OldRotation * DeltaRotation;
     DesiredMovement = (OldRotation * position);
 
+    // if Robot is on a moving platform, add the platform motion
+    if (MovingPlatform != nullptr && FollowFloor )
+    {
+        FVector CurrentPlatformLocation = FVector(MovingPlatform->GetActorLocation());
+        FVector PlatformTranslation = CurrentPlatformLocation - LastPlatformLocation;
+        DesiredMovement += PlatformTranslation;
+
+        // TODO : add platform rotation in DesiredRotation
+        
+        LastPlatformLocation = FVector(CurrentPlatformLocation);
+    }

```

Comment:
* `bFollowFloor` or `bFollowPlatform` for naming consistency
* Local vars with camelCase: `oldRotation, deltaRotation`
```suggestion
    if (MovingPlatform != nullptr && FollowFloor )
    {
        FVector currentPlatformLocation = MovingPlatform->GetActorLocation();
        DesiredMovement += (currentPlatformLocation - LastPlatformLocation);

        LastPlatformLocation = currentPlatformLocation;        
        // TODO : add platform rotation in DesiredRotation
        // -> Might require tracking MovingPlatform's last rotation to identify deltaPlatformRotation?
    }
```